hipogram/posts/views.py

```diff
@@ -39,41 +40,37 @@ def post_new(request):
         form = PostForm()
     return render(request, 'share.html', {'form': form})
     
-#to delete a post
-def delete_post(request, pk):
-    template_name = 'update.html'
-    post = get_object_or_404(Post, pk=pk)
+@login_required()
+def delete_post(request, id):
+    post = get_object_or_404(Post, id=id)
 
-    try:
+    if request.user == post.created_by:
         if request.method == 'POST':
             form = PostForm(request.POST, instance=post)
             post.delete()
             messages.success(request, 'You have successfully deleted the post')
```

Comment:
Using `messages` is great. Can you please add messages for update view too and also add fail messages for redirecting non-owners?